packages/node_modules/@node-red/registry/lib/util.js

```diff
@@ -79,7 +83,7 @@ function createNodeApi(node) {
         httpAdmin: runtime.adminApp,
         server: runtime.server
     }
-    copyObjectProperties(runtime.nodes,red.nodes,["createNode","getNode","eachNode","addCredentials","getCredentials","deleteCredentials" ]);
+    copyObjectProperties(runtime.nodes,red.nodes,["createNode","getNode","eachNode","addCredentials","getCredentials","deleteCredentials", "listNPMModules"]);
```

Comment:
Why is `listNPMModules` being exposed to nodes? They don't need to query that list... unless this is because you are exposing the list of modules from the Function node's file...

This feature is either entirely owned by the Function node, or it is a core feature that works even if the Function node is removed from the install.

If there is an admin HTTP route needed by the editor to provide admin of the module list, then it needs to be a core admin route.

Reply:
As commented above, I will consider moving this to admin API.